src/vsr/replica.zig

```diff
@@ -2094,19 +2094,14 @@ pub fn ReplicaType(
 
             const op_checkpoint_max =
                 DVCQuorum.op_checkpoint_max(self.do_view_change_from_all_replicas);
-            if (op_checkpoint_max > self.op_checkpoint() and op_head > self.op_prepare_max()) {
-                // When:
-                // 1. the cluster is at a checkpoint ahead of the local checkpoint,
-                // 2. AND the quorum's head op is part of a future checkpoint,
-                //    abdicate as primary, jumping to a new view.
-                //
-                // - If 1 and ¬2, then the ops in the next checkpoint are uncommitted and our
-                //   current checkpoint is sufficient.
-                // - If 2 and ¬1, then the ops past the checkpoint's op-prepare-max must be
-                //   uncommitted because no replica in the DVC quorum even prepared them.
-                //   (They are present anyway because the origin of the DVC is reusing its SV
-                //   headers as a DVC since it has not completed repair.)
-                //
+            const commit_max = @max(
+                self.commit_max,
+                DVCQuorum.commit_max(self.do_view_change_from_all_replicas),
+            );
+            // We are lagging from the cluster by at least a checkpoint, and that checkpoint is
+            // durable on a commit quorum of replicas. Forfeit the view change and prefer the
+            // replica with the durable checkpoint for primary.
+            if (commit_max > self.op_prepare_max()) {
```

Comment:
I guess we could look only at `DVCQuorum.commit_max(self.do_view_change_from_all_replicas),` here, we don't need to `@max(self.commit_max`?

actually, could we asser that `DVCQUorum.commit_max >= self.commit_max`? Our DVC should also be a part of the quorum I think?

Reply:
>actually, could we asser that DVCQUorum.commit_max >= self.commit_max? Our DVC should also be a part of the quorum I think?

RElevant code:

```
            // When computing the new commit_max, we cannot simply rely on the fact that our own
            // commit_min is attached to our own DVC in the quorum.
            // Consider the case:
            // 1. Start committing op=N…M.
            // 2. Send `do_view_change` to self.
            // 3. Finish committing op=N…M.
            // 4. Remaining `do_view_change` messages arrive, completing the quorum.
            // In this scenario, our own DVC's commit is `N-1`, but `commit_min=M`.
            // Don't let the commit backtrack.
            const commit_max = @max(
                self.commit_min,
                DVCQuorum.commit_max(self.do_view_change_from_all_replicas),
            );
```

So it looks like we should advance to `commit_min`,if we want to have this assert. But, as we don't have to advance, I think we could just get without this max at all!

Reply:
This PR also removes this block of code (https://github.com/tigerbeetle/tigerbeetle/pull/2236/files#diff-e95e4dc6c1aae349e798cfeed22ad01f830cce0e654e6b4b765b3e7b0c677b6aL8057-L8069), because we anyway take the max with self.commit_max in set_op_and_commit_max: https://github.com/tigerbeetle/tigerbeetle/blob/241c20fc3c3716ef694532568ddc64694e6e76e8/src/vsr/replica.zig#L7973-L7975